### DISCUSSION TITLE
XUnit share context without constructor injection

### DISCUSSION THREAD

**Discussion by NiceStepUp** [xunit/xunit#2597] [Question] [answered]
I am trying to share context between all tests. I've found [this beautiful tutorial how it can be done through xunit](https://xunit.net/docs/shared-context).

So I've created database fixture to be shared between all tests like the above tutorial said:

```
public interface IDatabase
{
    string Read();
}

public class DatabaseFixture : IDisposable, IDatabase
{
    public DatabaseFixture()
    {
        Db = new SqlConnection("MyConnectionString");

        // ... initialize data in the test database ...
    }

    public void Dispose()
    {
        // ... clean up test data from the database ...
    }

    public SqlConnection Db { get; private set; }
    public string Read()
    {
        throw new NotImplementedException();
    }
}
```
And:

```
[CollectionDefinition("Database collection")]
public class DatabaseCollection : ICollectionFixture<DatabaseFixture>
{
    // This class has no code, and is never created. Its purpose is simply
    // to be the place to apply [CollectionDefinition] and all the
    // ICollectionFixture<> interfaces.
}   
```
And shared fixture looks like this:

```
[Collection("Database collection")]
public class DatabaseBaseClass
{
    IDatabase _fixture;

    public DatabaseBaseClass(DatabaseFixture fixture)
    {
        _fixture = fixture;
    }
}
```
So far, so good, it works perfectly. However, I **have more than 300 classes and what I have to do is to write this constructor initialization in more than 300 classes of tests**:

```
public class PersonTests : DatabaseBaseClass
{
    protected PersonTests(DatabaseFixture fixture)
        : base(fixture)
    {}
}
```
My question is how is it possible to avoid to write the above constructor in more than 300 classes of tests? Is it possible? Maybe is there more simpler way?

**Comment by bradwilson** [comment, marked as answer]
You don't need to add the constructor if you don't need access to the fixture object from within the unit test. It will still be created appropriately.

If you do need access to the fixture object, then there's no way around this today. Constructor injection is the only way to get access to fixture objects in v2.
